section-05-production-model-package/regression_model/processing/validation.py

```diff
@@ -0,0 +1,129 @@
+from typing import Tuple, Optional, List
+from regression_model.config.core import config
+
+from pydantic import BaseModel, ValidationError
+import numpy as np
+import pandas as pd
+
+
+def drop_na_inputs(*, input_data: pd.DataFrame) -> pd.DataFrame:
+    """Check model inputs for na values and filter."""
+    validated_data = input_data.copy()
+    new_vars_with_na = [
+        var for var in config.model_config.features
+        if var not in config.model_config.categorical_vars_with_na_frequent +
+           config.model_config.categorical_vars_with_na_missing +
+           config.model_config.numerical_vars_with_na
+           and validated_data[var].isnull().sum() > 0
+    ]
+    validated_data.dropna(subset=new_vars_with_na, inplace=True)
+
+    return validated_data
+
+
+def validate_inputs(
+        *, input_data: pd.DataFrame
+) -> Tuple[pd.DataFrame, Optional[dict]]:
+    """Check model inputs for unprocessable values."""
+
+    # convert syntax error field names (beginning with numbers)
+    input_data.rename(columns=config.model_config.variables_to_rename, inplace=True)
+    input_data.drop('Id', axis=1, inplace=True)
```

Comment:
Here we are dropping Id, because unfortunately, it is in the test data. But, in real life, we wouldn't send a variable that we do not have to an API with a model. Can we not validate by selecting the features that will be used, instead of dropping those not needed?

Reply:
improved as suggested